pymc3/tests/test_parallel_sampling.py

```diff
@@ -172,24 +172,23 @@ def func(x):
         trace = pm.sample(draws=10, tune=10, step=pm.Metropolis(), cores=2, mp_ctx="spawn")
 
 
-@pytest.mark.xfail(raises=ValueError)
 def test_spawn_densitydist_bound_method():
     with pm.Model() as model:
         mu = pm.Normal("mu", 0, 1)
         normal_dist = pm.Normal.dist(mu, 1)
         obs = pm.DensityDist("density_dist", normal_dist.logp, observed=np.random.randn(100))
-        trace = pm.sample(draws=10, tune=10, step=pm.Metropolis(), cores=2, mp_ctx="spawn")
+        msg = "logp for DensityDist is a bound method, leading to RecursionError while serializing"
+        with pytest.raises(ValueError, match=msg):
```

Comment:
I considered `pytest.raises`, but the code in `DensityDist` is not *guaranteed* to raise the error. It will try to serialize as normal, and only *if* a `RecursionError` occurs (at the moment, expected to happen), it's wrapped and re-raised. It's possible that a later fix in `dill` or so will prevent the error from happening altogether. So I decided on `xfail` instead.

Reply:
ah, I see - thanks for explaining!

> It's possible that a later fix in dill or so will prevent the error from happening altogether

If that happens, maybe it's good that this test fails (i.e. doesn't raise a `ValueError`) because then we will get a loud and clear notice that the error no longer happens and then we can set a minimum `dill` version?

Reply:
Yep, sounds good to me :)